harness/api_checkers.go

```diff
@@ -0,0 +1,400 @@
+// Package harness provides a suite of API compatibility checks. They were originally
+// ported from the OpenTracing Python library's "harness" module.
+package harness
+
+import (
+	"bytes"
+	"fmt"
+	"testing"
+	"time"
+
+	opentracing "github.com/opentracing/opentracing-go"
+	"github.com/opentracing/opentracing-go/log"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/suite"
+)
+
+// APICheckCapabilities describes options used by APICheckSuite when testing a Tracer.
+type APICheckCapabilities struct {
+	CheckBaggageValues bool          // whether to check for propagation of baggage values
+	CheckExtract       bool          // whether to check if extracting contexts from carriers works
+	CheckInject        bool          // whether to check if injecting contexts works
+	Probe              APICheckProbe // optional interface providing methods to check recorded data
+}
+
+// APICheckProbe exposes methods for testing data recorded by a Tracer.
+type APICheckProbe interface {
+	SameTrace(first, second opentracing.Span) bool // whether two spans are from the same trace
+	SameTraceContext(opentracing.Span, opentracing.SpanContext) bool
```

Comment:
Originally I thought this would assert a span and context were simply part of the same trace, but looking at my test cases it is actually being used to assert an extracted context matches the span it came from after being serialized & deserialized, and comments.